### PR TITLE
Deprecate array sorted and reverse

### PR DESCRIPTION
**Pull request by stonea**
As described here: https://github.com/chapel-lang/chapel/issues/18089#issuecomment-1276500706

We've decided to deprecate array's `sorted` and `reverse` methods. These were previously marked unstable.

### CODE REVIEW

File: test/studies/shootout/submitted/knucleotide3.chpl [hunk 1]
```diff
@@ -57,7 +57,7 @@ proc writeFreqs(data, param nclSize) {
   var arr = for (s,f) in freqs.items() do (f,s);
 
   // print the array, sorted by decreasing frequency
-  for (f, s) in arr.sorted(reverseComparator) do
+  for (f, s) in sorted(arr, reverseComparator) do
```
Comment (lydia-duncan) [inline test/studies/shootout/submitted/knucleotide3.chpl:60]: Submitted shootouts should not be updated in their body until the code on the website has been updated, instead we update their .good files if we need to or mark them as .notest if they do not compile at all.  There's a README in this directory describing what to do

File: test/studies/shootout/submitted/knucleotide4.chpl [hunk 2]
```diff
@@ -56,7 +56,7 @@ proc writeFreqs(data, param nclSize) {
   var arr = for (s,f) in freqs.items() do (f,s.val);
 
   // print the array, sorted by decreasing frequency
-  for (f, s) in arr.sorted(reverseComparator) do
+  for (f, s) in sorted(arr, reverseComparator) do
```
Comment (lydia-duncan) [inline test/studies/shootout/submitted/knucleotide4.chpl:59]: Ditto here

**Review by lydia-duncan — APPROVED** [review]
Looks good aside from the note about the updated shootouts - I probably don't need to see the fixes for those but I'm happy to take a look again when they happen

File: test/studies/shootout/submitted/knucleotide3.chpl (hunk 1 above)
Comment (stonea) [reply test/studies/shootout/submitted/knucleotide3.chpl]: Whoah, good to know; I'm glad I got this reviewed.  I've reverted the change in these two files and updated the corresponding `.good` file to include the new deprecation messages

**Review by lydia-duncan — APPROVED** [review]
Thanks!

**Pull request merged**